### ISSUE TITLE
Smart Hyperparameter Search for PEFT

### ISSUE THREAD

**Issue by PamelaBha** [huggingface/transformers#40533]
### Feature request

Problem: Fine-tuning large models with Parameter-Efficient Fine-Tuning (PEFT) methods like LoRA requires choosing optimal hyperparameters (e.g., LoRA rank, alpha, dropout). Manually searching for the best combination is time-consuming and often leads to sub-optimal results.

### Motivation

 This would be a massive time-saver for users. Instead of running dozens of experiments, they could run a single command and get a recommended PEFT config that maximizes performance while minimizing trainable parameters. This aligns perfectly with the goal of improving the fine-tuning experience.

### Your contribution

I would work on the entire PR

**Comment by PamelaBha** [comment]
cc @Rocketknight1 - what do you think of this?

**Comment by Rocketknight1** [comment]
cc @sayakpaul @BenjaminBossan for PEFT!

**Comment by BenjaminBossan** [comment]
Hmm, I'm not sure yet if I understand the scope of this suggestion. In general, it should already be possible to search the hyperparameter space of transformers models using existing packages, e.g. [optuna](https://huggingface.co/learn/cookbook/en/optuna_hpo_with_transformers). Adding PEFT to the mix does not really change the picture IMO. If you prefer scikit learn's hyperparameter search features, you can combine transformers models with [skorch](https://github.com/skorch-dev/skorch). Therefore, unless I misunderstand the suggestion, I'd say this is not necessary to add to transformers.

I agree that it can be a real issue finding the optimal hyperparameters for PEFT. This is why we're working on the [method comparison suite](https://github.com/huggingface/peft/tree/main/method_comparison). This should help users make better choices when it comes to which PEFT method or which setting best aligns with their desired trade offs.

**Comment by PamelaBha** [comment]
Got it. I wasn't aware of the comparison suite.
